GVFS/GVFS.Platform.Windows/WindowsFileSystem.cs

```diff
@@ -29,6 +31,39 @@ public void ChangeMode(string path, int mode)
         {
         }
 
+        public bool IsPathUnderDirectory(string directoryPath, string path)
+        {
+            // TODO: Normalize paths
+            // We can't use the existing TryGetNormalizedPathImplementation method
+            // because it relies on actual calls to the disk to check if directories exist.
+            // This may be called with paths or volumes which do not actually exist.
+            return path.StartsWith(directoryPath, StringComparison.OrdinalIgnoreCase);
+        }
+
+        public string GetVolumeRoot(string path)
+        {
+            var volumePathName = new StringBuilder(GVFSConstants.MaxPath);
+            if (NativeMethods.GetVolumePathName(path, volumePathName, GVFSConstants.MaxPath))
+            {
+                return volumePathName.ToString();
+            }
+
+            return null;
+        }
+
+        public bool IsVolumeAvailable(string path)
+        {
+            // No paths 'exist' on locked BitLocker volumes so it is sufficent to just
+            // check if the directory/file exists using the framework APIs.
+            return System.IO.Directory.Exists(path) || System.IO.File.Exists(path);
+        }
+
+        public IVolumeStateWatcher CreateVolumeStateWatcher()
+        {
+            // TODO: Extract the polling interval to a configuration value?
+            return new WindowsVolumeStateWatcher(TimeSpan.FromSeconds(15));

```

Comment:
1. If we are going to be polling anyways, wouldn't it be simpler to just iterate though the yet-to-be-mounted repository list and check if the Enlistment root path became available and readable in the Timer callback? Thats all that we need to know, right? Or does the volumeWatcher and cryptoVolumeWatcher offer anything more?

2. After all the un-available repos get mounted eventually, do we shutdown this timer?